### ISSUE TITLE
Problem tab issue

### ISSUE THREAD

**Issue by pabloto** [codefori/vscode-ibmi#1678]
I'm working in local, my Ibm i have an iasp.

If I compile a program in tab output I see twice group with my program name, and if I click on a  compilation error it open me an empty tab:

<img width="1316" alt="image" src="https://github.com/codefori/vscode-ibmi/assets/9719313/20e991f9-044d-4bda-b3b4-881f420c5454">

<img width="1396" alt="image" src="https://github.com/codefori/vscode-ibmi/assets/9719313/8d844fe9-f691-4d10-9120-1b329a926cbd">

Let me know if I have to share something else.

Thank you

Bye

<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.4.0|
|Visual Studio Code version|1.84.2|
|Operating System|darwin_x64|

<details>
<summary>Active extensions</summary>

```
COBOL (cobol): 9.11.22
Code Spell Checker (code-spell-checker): 3.0.1
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.4.0
Db2 for IBM i (vscode-db2i): 0.5.2
Dev Containers (remote-containers): 0.321.0
ESLint (vscode-eslint): 2.4.2
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.15.0
Extension Authoring (extension-editing): 1.0.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
Git Graph (git-graph): 1.30.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
GitLab Workflow (gitlab-workflow): 3.86.1
GitLens — Git supercharged (gitlens): 14.5.1
IBM i Project Explorer (vscode-ibmi-projectexplorer): 1.2.6
JSON Language Features (json-language-features): 1.0.0
Live Share (vsliveshare): 1.0.5892
Merge Conflict (merge-conflict): 1.0.0
Microsoft Account (microsoft-authentication): 0.0.1
NPM support for VS Code (npm): 1.0.1
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Postman (postman-for-vscode): 0.14.3
RPGLE (vscode-rpgle): 0.24.0
Rainbow CSV (rainbow-csv): 3.8.0
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
WSL (remote-wsl): 0.81.8
vscode-icons (vscode-icons): 12.6.0
```
</details>

<hr />

**Comment by worksofliam** [comment]
Odd. Are you able to help us reproduce this problem with an example repo and some details about your system?

**Comment by pabloto** [comment]
<img width="928" alt="image" src="https://github.com/codefori/vscode-ibmi/assets/9719313/5f4b78d9-4c09-477a-a341-3d78a6504448">

as you can see the output tab add the name of my iasp: directa1

**Comment by worksofliam** [comment]
I suspect what is happening is that we are not taking in account the ASP when compiling from local.

Code around here I think: https://github.com/codefori/vscode-ibmi/blob/master/src/api/errors/diagnostics.ts#L143

**Comment by worksofliam** [comment]
@pabloto Would you be able to share what your deploy directory is? You can see the path when selecting the deployment method: 

<img width="618" alt="image" src="https://github.com/codefori/vscode-ibmi/assets/3708366/2b399162-7519-43d0-86ae-11ac3e34ae5f">

I need to compare it to what the compiler is giving us.

**Comment by worksofliam** [comment]
@pabloto I also need a copy of the member in `EVFEVENT` when after the compile happens with the errors, so it shows me the full path it uses.

The output log will have a command like this after the compile, which will show the `EVFEVENT` member: 

```
/home/paolos/builds/test_tab_problem: system "QSYS/CPYTOIMPF FROMFILE(PAOLOS/EVFEVENT TESTTIMEST) TOSTMF('/tmp/vscodetemp-O_ZlwgXIlN') MBROPT(*REPLACE) STMFCCSID(1208) RCDDLM(*CRLF) DTAFMT(*DLM) RMVBLANK(*TRAILING) ADDCOLNAM(*SQL) FLDDLM(',') DECPNT(*PERIOD)"
{
    "code": 0,
    "signal": null,
    "stdout": "CPC2958: Tutti i record copiati dal file EVFEVENT in PAOLOS",
    "stderr": ""
}
```

**Comment by pabloto** [comment]
> @pabloto Would you be able to share what your deploy directory is? You can see the path when selecting the deployment method:
> 
> <img alt="image" width="618" src="https://private-user-images.githubusercontent.com/3708366/295004107-2b399162-7519-43d0-86ae-11ac3e34ae5f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3OTc4NjAsIm5iZiI6MTcwNDc5NzU2MCwicGF0aCI6Ii8zNzA4MzY2LzI5NTAwNDEwNy0yYjM5OTE2Mi03NTE5LTQzZDAtODZhZS0xMWFjM2UzNGFlNWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDlUMTA1MjQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2Y4NmFhODdmNWM4OTA0ODcxNWNlN2ExYmJlYjMxYWE2MjhhN2VjZTUyMGU5MTNkM2Q5MmU1NDI2MTliOWRmNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.0svugEu84XsSZKaaP9G7lSTU3OChenNxnipuOdeKy1U">
> I need to compare it to what the compiler is giving us.

Hi Liam here's the screenshot about deploy directory:

<img width="1369" alt="image" src="https://github.com/codefori/vscode-ibmi/assets/9719313/0f8cfb98-662d-4e7c-8c43-0384a2ec33a4">

**Comment by pabloto** [comment]
> @pabloto I also need a copy of the member in `EVFEVENT` when after the compile happens with the errors, so it shows me the full path it uses.
> 
> The output log will have a command like this after the compile, which will show the `EVFEVENT` member:
> 
> ```
> /home/paolos/builds/test_tab_problem: system "QSYS/CPYTOIMPF FROMFILE(PAOLOS/EVFEVENT TESTTIMEST) TOSTMF('/tmp/vscodetemp-O_ZlwgXIlN') MBROPT(*REPLACE) STMFCCSID(1208) RCDDLM(*CRLF) DTAFMT(*DLM) RMVBLANK(*TRAILING) ADDCOLNAM(*SQL) FLDDLM(',') DECPNT(*PERIOD)"
> {
>     "code": 0,
>     "signal": null,
>     "stdout": "CPC2958: Tutti i record copiati dal file EVFEVENT in PAOLOS",
>     "stderr": ""
> }
> ```

Hi @worksofliam,

here's the output of cpyfrmstmf:

`
/home/paolos: /QOpenSys/usr/bin/qsh
system  "QSYS/CPYTOIMPF FROMFILE(PAOLOS/EVFEVENT TESTTIMEST) TOSTMF('/tmp/vscodetemp-O_K3SSpaFF') MBROPT(*REPLACE) STMFCCSID(1208) RCDDLM(*CRLF) DTAFMT(*DLM) RMVBLANK(*TRAILING) ADDCOLNAM(*SQL) FLDDLM(',') DECPNT(*PERIOD)"
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": "CPC2206:  È stata modificata la proprietà dell'oggetto QCPEXTEMPS in QTEMP di tipo *USRSPC.\nCPC2206:  È stata modificata la proprietà dell'oggetto QACP646345 in QTEMP di tipo *USRSPC.\nCPC2206:  È stata modificata la proprietà dell'oggetto QACEXFLD12 in QTEMP di tipo *USRSPC.\nCPC2206:  È stata modificata la proprietà dell'oggetto Q1T2TT02 in QTEMP di tipo *FILE.\nCPC2958:  Tutti i record copiati dal file EVFEVENT in PAOLOS\nCPC2206:  È stata modificata la proprietà dell'oggetto QZSHSYSTEM in QTEMP di tipo *USRSPC."
}
`

the evfevent file:



<img width="2147" alt="image" src="https://github.com/codefori/vscode-ibmi/assets/9719313/71d2dac3-3229-4260-9270-c6bd1c003294">

Has you can see in problems tab I see my program testtimest.rpgle twice:

1. the first testtimest file is linked to directa1/qrpglesrc
2. the second testtimest file is linked to qrpglesrc

Let me know if I can send you more informations.

Thank you.

**Comment by worksofliam** [comment]
@pabloto Thanks!! That has all the info I need! 

Even though you are deploying to `/home/paolos/...`, the compiler is giving `/DIRECTA1/home/paolos/...`. This should be doable. Thanks!

**Comment by pabloto** [comment]
really /home is a symbolic link to /DIRECTA1/home
if it could help.

Thanks!

**Comment by worksofliam** [comment]
The plan is simple to check the compile error path to see if it starts with an ASP name and substitute it from the path.

**Comment by worksofliam** [comment]
Awesome, thank you!!

**Comment by worksofliam** [comment]
@pabloto This has been released in 2.6.5! Thanks!
